alws/crud/release.py

```diff
@@ -19,24 +19,57 @@
 
 
 async def get_releases(
-        page_number: typing.Optional[int],
-        db: Session
-) -> typing.List[models.Release]:
-    query = select(models.Release).options(
-        selectinload(models.Release.owner),
-        selectinload(models.Release.platform),
-        selectinload(models.Release.product),
-    ).order_by(models.Release.id.desc())
-    if page_number:
-        query = query.slice(10 * page_number - 10, 10 * page_number)
-    release_result = await db.execute(query)
+    db: Session,
+    page_number: typing.Optional[int] = None,
+    release_id: int = None,
+    search_params: release_schema.ReleaseSearch = None,
+) -> typing.Union[typing.List[models.Release], models.Release]:
+
+    def generate_query(count=False):
+        query = select(models.Release).options(
+            selectinload(models.Release.owner),
+            selectinload(models.Release.platform),
+            selectinload(models.Release.product),
+        ).order_by(models.Release.id.desc())
+        if count:
+            query = select(func.count(models.Release.id))
+        if release_id:
+            query = query.where(models.Release.id == release_id)
+        if search_params:
+            # These links could be helpful for filtering by JSON
+            # https://github.com/sqlalchemy/sqlalchemy/discussions/7991
+            # https://www.postgresql.org/docs/9.5/functions-json.html
+            # if search_params.package_name:
```

Comment:
Should it be commented?

Reply:
I comment it by purpose, I don't have enough time right now for researching how to make a query for JSON field, so I provide some links for the future